### PR TITLE
Manage the state of the product depending on quantity in cart

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Before, you could add more product in cart than you were able to
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/27144.
| How to test?      | Use a product with only 1 quantity remaining and oosp disabled, try to add it to the cart from a wishlist, you should be able, but after the product addToCart button should be disabled
| Possible impacts? | Add to cart from wishlist

## Other cases fixed by this PR
In case the product has a minimal quantity
In case the product has out of stock order allowed or not

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

I found an issue following these steps:
1. Go to FO > Sign in with a customer
2. Go to the product page
3. Increase the quantity to 5
4. Click on the heart and add the product to the wishlist
5. Go to FO > Home >  Your account  > My wishlists
6. The Add to cart is disabled, it shouldn't 

image

Thanks!

**Comment by NeOMakinG** [comment]
@khouloudbelguith Damn, I used the wrong key for the quantity check, it should be fine now

File: _dev/front/js/components/Product/Product.vue [hunk 1]
```diff
@@ -226,10 +226,25 @@
     },
     computed: {
       isDisabled() {
+        console.log(this.product);
```
Comment (PierreRambaud) [inline _dev/front/js/components/Product/Product.vue:229]:
```suggestion
```

Comment (NeOMakinG) [reply _dev/front/js/components/Product/Product.vue:229]: Damn...

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

Please check the Conflicting files.

Thanks!

**Review by PierreRambaud — APPROVED** [review]

**Comment by NeOMakinG** [comment]
> Hi @NeOMakinG,
> 
> I have two issues: I checked this module with 178x branch.
> 
> **First issue**
> 
> 1. Create a product `ProductA`
> 2. Add a `productA` to the cart
> 3. Go to FO > View `productA` > And add it to the wishlist
> 4. Go to FO > Home > Your account > My wishlists
> 
> See error: The Add to cart is disabled, it shouldn't
> 
>  Untitled_.Feb.10.2022.5_57.PM.mp4 
> So if we have the same product on a different wishlist, it will be added only from a list.
> 
> **Issue2** Reproduced only with demo products
> 
> 1. Go to FO > Add the first product to mywishlist
> 2. Go to FO > Home > Your account > My wishlists
> 3. Click on add to the cart
> 4. Click on continue shopping
> 5. See error => the Add to cart is not disabled
> 
>  Untitled_.Feb.10.2022.6_09.PM.mp4 
> Thanks!

Did you change something to the number of products you have? I can't manage to reproduce at all

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

I tried again with fresh install 178x & PS1783.
I have the same behavior.
I attached a screen record

https://user-images.githubusercontent.com/16067358/153843307-653c3a22-a62f-492f-b84e-8ab4f166d596.mp4

Thanks to check and feedback.

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by khouloudbelguith — CHANGES_REQUESTED** [review]
Hi @NeOMakinG,

I have two issues:

**First issue**
1. Go to FO and sign in with a customer
2. Click on any product & increase the quantity
3. Add it to the wishlist
4. Go Your account > Wishlist list
5. Add the product to the cart & click on continue shopping

https://user-images.githubusercontent.com/16067358/157054719-c4435377-1ee4-4132-8b22-9a5e3dbd783e.mp4

The quantity is reset to the minimal quantity 

**Without PR**

https://user-images.githubusercontent.com/16067358/157055155-fbfb3e8c-892e-47eb-92d9-e5e35b513409.mp4

Ping @PrestaShop/product-team what do you think?

**Second issue**

1. Create a product out of stock allowed
2. Go to FO > Add this product to the wishlist
3. Go Your account > Wishlist list
5. Add the product to the cart & click on continue shopping
6. See error: the button add to cart disabled

https://user-images.githubusercontent.com/16067358/157056353-a4a751e1-d775-46ed-93e4-88a5eeb01349.mp4

Thanks!

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by khouloudbelguith — APPROVED** [review]
Hi @NeOMakinG,

It is ok :heavy_check_mark: 
I checked with develop branch & 178x branch.
I checked with:
simple product & quantity = 1
simple product & quantity > 1
product with combinations & minimal quantity > 1
pack product & quantity = decrement both
Vector product with quantity out of stock allowed

I checked with shop 1 & shop2

https://user-images.githubusercontent.com/16067358/162178193-cda12314-a03d-429d-bbe1-d2391821bbe6.mp4


https://user-images.githubusercontent.com/16067358/162179675-8fc60033-02ac-4d97-87ba-45a342200c06.mp4

Thanks!

**Comment by kpodemski** [comment]
🎉 

Thank you @NeOMakinG @khouloudbelguith

**Pull request merged**